curvefs/src/metaserver/copyset/copyset_node.cpp

```diff
@@ -540,8 +544,22 @@ void CopysetNode::ListPeers(std::vector<Peer>* peers) const {
     }
 }
 
-std::list<PartitionInfo> CopysetNode::GetPartitionInfoList() {
-    return metaStore_->GetPartitionInfoList();
+// if copyset is loading, return false;
+// if copyset is not loading, and metastore returns fales, retry;
+// if copyset is not loading, and metastore returns true, return true and get
+// partition info list success.
+bool CopysetNode::GetPartitionInfoList(
+                    std::list<PartitionInfo> *partitionInfoList) {
+    while (true) {
```

Comment:
Does the infinite loop logic is ok when metaStore_->GetPartitionInfoList failed？

Reply:
Under normal circumstances, one retry can be successful.